packages/kit/src/runtime/server/index.js

```diff
@@ -18,12 +18,19 @@ const default_transform = ({ html }) => html;
 export async function respond(request, options, state) {
 	let url = new URL(request.url);
 
-	if (
-		options.csrf.check_origin &&
-		request.method === 'POST' &&
-		request.headers.get('origin') !== url.origin
-	) {
-		return new Response(`Cross-site ${request.method} requests are forbidden`, { status: 403 });
+	if (options.csrf.check_origin) {
+		const type = request.headers.get('content-type');
+
+		const forbidden =
+			request.method === 'POST' &&
+			request.headers.get('origin') !== url.origin &&
+			(type === 'application/x-www-form-urlencoded' || type === 'multipart/form-data');
```

Comment:
I think these might have a `; charset=` in them that would throw off this check.

Reply:
I've never seen a `charset=`, but we _do_ need to split on `;` anyway because `multipart-formdata` will always be accompanied by `; boundary=`